src/ucp/api/ucp.h

```diff
@@ -333,6 +335,19 @@ typedef enum {
 
 /**
  * @ingroup UCP_DATATYPE
+ * @brief Generate a reusable identifier for Scatter-gather IOV data type.
```

Comment:
I don't mind making it transparent, but this means that we change the default behaviour of all datatypes to be memory allocations (to store this optimization), requiring free/destroy eventually, which breaks backward compatibility. Today create_iov() is a macro returning a constant value, and no free/destroy called on it. Am O missing something here?